app/src/org/commcare/activities/CommCareWiFiDirectActivity.java

```diff
@@ -637,9 +708,9 @@ protected void deliverError(CommCareWiFiDirectActivity receiver, Exception e) {
             }
 
             @Override
-            protected void deliverResult(CommCareWiFiDirectActivity receiver, FormRecord[] result) {
-                if (result != null) {
-                    receiver.onZipSuccesful(result);
+            protected void deliverResult(CommCareWiFiDirectActivity receiver, Integer result) {
+                if (result.intValue() == ZipTask.RESULT_SUCCESS) {

```

Comment:
Can you make `ZipTask` return an `enum` of possible outcomes instead of an `Integer` flag? This will make it easier to ensure all outcomes are handled and equality checks of the flags won't need to use unboxing because enum instances are always singletons; allowing for reference equality.